mutation_partition_view.cc

```diff
@@ -230,11 +230,107 @@ void mutation_partition_view::do_accept(const column_mapping& cm, Visitor& visit
     }
 }
 
+void mutation_partition_view::accept_ordered(const schema& s, mutation_partition_view_virtual_visitor& visitor) const {
+    auto in = _in;
+    auto mpv = ser::deserialize(in, boost::type<ser::mutation_partition_view>());
+    const column_mapping& cm = s.get_column_mapping();
+
+    visitor.accept_partition_tombstone(mpv.tomb());
+
+    struct static_row_cell_visitor {
+        mutation_partition_view_virtual_visitor& _visitor;
+
+        void accept_atomic_cell(column_id id, atomic_cell ac) const {
+           _visitor.accept_static_cell(id, std::move(ac));
+        }
+        void accept_collection(column_id id, const collection_mutation& cm) const {
+           _visitor.accept_static_cell(id, cm);
+        }
+    };
+    read_and_visit_row(mpv.static_row(), cm, column_kind::static_column, static_row_cell_visitor{visitor});
+
+    auto rt_it = mpv.range_tombstones().cbegin();
+    auto rt_e = mpv.range_tombstones().cend();
+    auto cr_it = mpv.rows().cbegin();
+    auto cr_e = mpv.rows().cend();
+
+    auto consume_rt = [&] (range_tombstone&& rt) {
+        visitor.accept_row_tombstone(std::move(rt));
+    };
+    auto consume_cr = [&] (ser::deletable_row_view&& cr) {
+        auto t = row_tombstone(cr.deleted_at(), shadowable_tombstone(cr.shadowable_deleted_at()));
+        visitor.accept_row(position_in_partition_view::for_key(cr.key()), t, read_row_marker(cr.marker()), is_dummy::no, is_continuous::yes);
+
+        struct cell_visitor {
+            mutation_partition_view_virtual_visitor& _visitor;
+
+            void accept_atomic_cell(column_id id, atomic_cell ac) const {
+               _visitor.accept_row_cell(id, std::move(ac));
+            }
+            void accept_collection(column_id id, const collection_mutation& cm) const {
+               _visitor.accept_row_cell(id, cm);
+            }
+        };
+        read_and_visit_row(cr.cells(), cm, column_kind::regular_column, cell_visitor{visitor});
+    };
+
+    std::optional<range_tombstone> rt;
+    auto next_rt = [&] {
+        if (!rt) {
+            if (rt_it != rt_e) {
+                rt = *rt_it;
+                ++rt_it;
+            }
+        }
+    };
+
+    std::optional<ser::deletable_row_view> cr;
+    auto next_cr = [&] {
+        while (!cr) {
+            if (cr_it != cr_e) {
+                cr = *cr_it;
+                ++cr_it;
+                auto ck = cr->key();
+                if (ck == clustering_key(std::vector<bytes>{})) {
```

Comment:
Use `is_empty()` instead, building an empty key is not free.

Reply:
Fixed in v2